plasmapy/diagnostics/thomson.py

```diff
@@ -177,24 +449,17 @@ def spectral_density(
             continue
         ion_species[ii] = Particle(ion)
 
-    # Condition Te
-    if Te.size == 1:
-        # If a single quantity is given, put it in an array so it's iterable
-        # If Te.size != len(efract), assume same temp. for all species
-        Te = np.repeat(Te, len(efract))
-    elif Te.size != len(efract):
-        raise ValueError(
-            f"Got {Te.size} electron temperatures and expected {len(efract)}."
-        )
-
     # Condition Ti
     if Ti.size == 1:
         # If a single quantity is given, put it in an array so it's iterable
         # If Ti.size != len(ion_species), assume same temp. for all species
-        Ti = [Ti.value] * len(ion_species) * Ti.unit
-    elif Ti.size != len(ion_species):
-        raise ValueError(
-            f"Got {Ti.size} ion temperatures and expected {len(ion_species)}."
+        Ti = (
+            np.array(
+                [
+                    Ti.value,
+                ]
+            )
+            * Ti.unit
```

Comment:
```suggestion
        Ti = np.array([Ti.value]) * Ti.unit
```

It's the comma after `Ti.value` that makes black want to break the expression across multiple lines.